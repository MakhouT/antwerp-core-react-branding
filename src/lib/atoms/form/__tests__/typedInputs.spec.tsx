import * as React from "react";

import * as A from "../../..";
import { renderAndMatchSnapshot } from "../../../../__tests__/__fixtures__/testUtils";

describe("TypedInputs", () => {
    it("ColorInput renders default", () => {
        // Arrange
        const element = <A.ColorInput
            name="input"
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("DateInput renders default", () => {
        // Arrange
        const element = <A.DateInput
            name="input"
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("DateTimeInput renders default", () => {
        // Arrange
        const element = <A.DateTimeInput
            name="input"
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("DateTimeInput renders default", () => {
        // Arrange
        const element = <A.DateTimeInput
            name="input"
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("DatetimeLocal renders default", () => {
        // Arrange
        const element = <A.DatetimeLocal
            name="input"
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("EmailInput renders default", () => {
        // Arrange
        const element = <A.EmailInput
            name="input"
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("NumberInput renders default", () => {
        // Arrange
        const element = <A.NumberInput
            name="input"
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("PasswordInput renders default", () => {
        // Arrange
        const element = <A.PasswordInput
            name="input"
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("TelInput renders default", () => {
        // Arrange
        const element = <A.TelInput
            name="input"
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("TextArea renders default", () => {
        // Arrange
        const element = <A.TextArea
            name="input"
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("TextInput renders default", () => {
        // Arrange
        const element = <A.TextInput
            name="input"
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("TimeInput renders default", () => {
        // Arrange
        const element = <A.TimeInput
            name="input"
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("UrlInput renders default", () => {
        // Arrange
        const element = <A.UrlInput
            name="input"
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("WeekInput renders default", () => {
        // Arrange
        const element = <A.WeekInput
            name="input"
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });
});
