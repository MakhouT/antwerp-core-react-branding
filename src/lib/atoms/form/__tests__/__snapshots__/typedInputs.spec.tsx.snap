// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypedInputs ColorInput renders default 1`] = `
<div
  className="a-input"
>
  <div
    className="a-input__wrapper"
  >
    <input
      id="-color-input"
      name="input"
      type="color"
    />
  </div>
  <small
    className=""
  >
     
  </small>
</div>
`;

exports[`TypedInputs DateInput renders default 1`] = `
<div
  className="a-input"
>
  <div
    className="a-input__wrapper"
  >
    <input
      id="-date-input"
      name="input"
      type="date"
    />
  </div>
  <small
    className=""
  >
     
  </small>
</div>
`;

exports[`TypedInputs DateTimeInput renders default 1`] = `
<div
  className="a-input"
>
  <div
    className="a-input__wrapper"
  >
    <input
      id="-datetime-input"
      name="input"
      type="datetime"
    />
  </div>
  <small
    className=""
  >
     
  </small>
</div>
`;

exports[`TypedInputs DateTimeInput renders default 2`] = `
<div
  className="a-input"
>
  <div
    className="a-input__wrapper"
  >
    <input
      id="-datetime-input"
      name="input"
      type="datetime"
    />
  </div>
  <small
    className=""
  >
     
  </small>
</div>
`;

exports[`TypedInputs DatetimeLocal renders default 1`] = `
<div
  className="a-input"
>
  <div
    className="a-input__wrapper"
  >
    <input
      id="-datetime-local-input"
      name="input"
      type="datetime-local"
    />
  </div>
  <small
    className=""
  >
     
  </small>
</div>
`;

exports[`TypedInputs EmailInput renders default 1`] = `
<div
  className="a-input"
>
  <div
    className="a-input__wrapper"
  >
    <input
      id="-email-input"
      name="input"
      type="email"
    />
  </div>
  <small
    className=""
  >
     
  </small>
</div>
`;

exports[`TypedInputs NumberInput renders default 1`] = `
<div
  className="a-input"
>
  <div
    className="a-input__wrapper"
  >
    <input
      id="-number-input"
      name="input"
      type="number"
    />
  </div>
  <small
    className=""
  >
     
  </small>
</div>
`;

exports[`TypedInputs PasswordInput renders default 1`] = `
<div
  className="a-input"
>
  <div
    className="a-input__wrapper"
  >
    <input
      id="-password-input"
      name="input"
      type="password"
    />
  </div>
  <small
    className=""
  >
     
  </small>
</div>
`;

exports[`TypedInputs TelInput renders default 1`] = `
<div
  className="a-input"
>
  <div
    className="a-input__wrapper"
  >
    <input
      id="-tel-input"
      name="input"
      type="tel"
    />
  </div>
  <small
    className=""
  >
     
  </small>
</div>
`;

exports[`TypedInputs TextArea renders default 1`] = `
<div
  className="a-input"
>
  <div
    className="a-input__wrapper"
  >
    <textarea
      id="-textarea-input"
      name="input"
      type="textarea"
    />
  </div>
  <small
    className=""
  >
     
  </small>
</div>
`;

exports[`TypedInputs TextInput renders default 1`] = `
<div
  className="a-input"
>
  <div
    className="a-input__wrapper"
  >
    <input
      id="-text-input"
      name="input"
      type="text"
    />
  </div>
  <small
    className=""
  >
     
  </small>
</div>
`;

exports[`TypedInputs TimeInput renders default 1`] = `
<div
  className="a-input"
>
  <div
    className="a-input__wrapper"
  >
    <input
      id="-time-input"
      name="input"
      type="time"
    />
  </div>
  <small
    className=""
  >
     
  </small>
</div>
`;

exports[`TypedInputs UrlInput renders default 1`] = `
<div
  className="a-input"
>
  <div
    className="a-input__wrapper"
  >
    <input
      id="-url-input"
      name="input"
      type="url"
    />
  </div>
  <small
    className=""
  >
     
  </small>
</div>
`;

exports[`TypedInputs WeekInput renders default 1`] = `
<div
  className="a-input"
>
  <div
    className="a-input__wrapper"
  >
    <input
      id="-week-input"
      name="input"
      type="week"
    />
  </div>
  <small
    className=""
  >
     
  </small>
</div>
`;
